packages/protovalidate/src/planner.ts

```diff
@@ -155,11 +155,21 @@ export class Planner {
     );
   }
 
-  private fields(fields: DescField[]): Eval<ReflectMessage> {
+  private fields(
+    msgRules: MessageRules,
+    fields: DescField[],
+  ): Eval<ReflectMessage> {
     const evals = new EvalMany<ReflectMessage>();
     for (const field of fields) {
       const fieldRules = getOption(field, ext_field);
-      if (fieldRules.required && fieldRules.ignore !== Ignore.ALWAYS) {
+      let ignore = fieldRules.ignore;
+      if (
+        !isFieldSet(fieldRules, field) &&
```

Comment:
This doesn't look right. Should that be `isFieldSet(fieldRules, FieldRulesSchema.field.ignore)`?

We should have conformance tests for this behavior.